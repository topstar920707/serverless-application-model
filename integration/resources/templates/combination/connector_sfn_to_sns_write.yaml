Resources:
  TriggerStateMachine:
    Type: AWS::Serverless::StateMachine
    Properties:
      Type: EXPRESS
      Definition:
        StartAt: TryDoSomething
        States:
          TryDoSomething:
            Type: Task
            Resource: arn:aws:states:::sns:publish
            Parameters:
              TopicArn: !Ref MyTopic
              Message: hello
            End: True
      Policies:
        - LambdaInvokePolicy:
            FunctionName: "SomethingIsNotStateMachineFunction"

  MyTopic:
    Type: AWS::SNS::Topic

  MyConnector:
    Type: AWS::Serverless::Connector
    Properties:
      Source:
        Id: TriggerStateMachine
      Destination:
        Id: MyTopic
      Permissions:
        - Write
