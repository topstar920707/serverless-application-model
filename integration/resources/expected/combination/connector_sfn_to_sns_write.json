[
  {
    "LogicalResourceId": "TriggerStateMachineRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "TriggerStateMachine",
    "ResourceType": "AWS::StepFunctions::StateMachine"
  },
  {
    "LogicalResourceId": "MyTopic",
    "ResourceType": "AWS::SNS::Topic"
  },
  {
    "LogicalResourceId": "MyConnectorPolicy",
    "ResourceType": "AWS::IAM::ManagedPolicy"
  }
]